lib/datasource.js

```diff
@@ -375,6 +375,14 @@ DataSource.prototype.setup = function(name, settings) {
   name = name || (connector && connector.name);
   this.name = name;
 
+  if (typeof settings === 'object' && typeof settings.name === 'string' && name !== settings.name) {
+    console.warn(
+      'A datasource has a name of %s while a name of %s is specified in ' +
```

Comment:
Minor nit: Can you wrap the formatters in quotes?
`A datasource has a name of "%s" while a name of "%s" is specified in ' + `

Reply:
+1, this is a good idea